src/hidimstat/desparsified_lasso.py

```diff
@@ -152,271 +122,382 @@ def desparsified_lasso(
+    P = (np.dot(X_.T, Z) / np.sum(X_ * Z, axis=0)).T
     P_nodiag = P - np.diag(np.diag(P))
     Id = np.identity(n_features)
     P_nodiag = dof_factor * P_nodiag + (dof_factor - 1) * Id
-
-    beta_hat = beta_bias - P_nodiag.dot(beta_lasso)
-
+    beta_hat = beta_bias.T - P_nodiag.dot(beta_reid.T)
+    # confidence intervals
     omega_diag = omega_diag * dof_factor**2
-    omega_invsqrt_diag = omega_diag ** (-0.5)
 
+    if not group:
+        return beta_hat, sigma_hat, omega_diag
+    else:
+        cov_hat = sigma_hat
+        if cov is not None:
+            cov_hat = cov
+        theta_hat = n_samples * inv(cov_hat)
+        return beta_hat, theta_hat, omega_diag
+
+
+def desparsified_lasso_pvalue(
+    n_samples,
+    beta_hat,
+    sigma_hat,
+    omega_diag,
+    confidence=0.95,
+    distribution="norm",
+    eps=1e-14,
+    confidence_interval_only=False,
+):
+    """
+    Calculate confidence intervals and p-values for desparsified lasso estimators.
+    This function computes confidence intervals for the desparsified lasso
+    estimator beta_hat.
+    It can also return p-values derived from these confidence intervals.
+    Parameters
+    ----------
+    n_samples : float
+        The number of samples
+    beta_hat : ndarray, shape (n_features,)
+        The desparsified lasso coefficient estimates.
+    sigma_hat : float
+        Estimated noise level.
+    omega_diag : ndarray, shape (n_features,)
+        Diagonal elements of the precision matrix estimate.
+    confidence : float, default=0.95
+        Confidence level for intervals, must be in [0, 1].
+    distribution : str, default="norm"
+        Distribution to use for p-value calculation.
+        Currently only "norm" supported.
+    eps : float, default=1e-14
+        Small value to avoid numerical issues in p-value calculation.
+    confidence_interval_only : bool, optional (default=False)
+        If True, return only confidence intervals.
+        If False, also return p-values.
+    Returns
+    -------
+    If confidence_interval_only=True:
+        cb_min : ndarray, shape (n_features,)
+            Lower bounds of confidence intervals
+        cb_max : ndarray, shape (n_features,)
+            Upper bounds of confidence intervals
+    If confidence_interval_only=False:
+        pval : ndarray, shape (n_features,)
+            P-values
+        pval_corr : ndarray, shape (n_features,)
+            Corrected p-values
+        one_minus_pval : ndarray, shape (n_features,)
+            1 - p-values
+        one_minus_pval_corr : ndarray, shape (n_features,)
+            1 - corrected p-values
+        cb_min : ndarray, shape (n_features,)
+            Lower bounds of confidence intervals
+        cb_max : ndarray, shape (n_features,)
+            Upper bounds of confidence intervals
+    """
+    # define the quantile for the confidence intervals
     quantile = stats.norm.ppf(1 - (1 - confidence) / 2)
-
+    # TODO:why the double inverse of omega_diag?
```

Comment:
3. The calculation of confint_radius inverse 2 times the omega_diag. I don't see the reason of it.

Reply:
because the 95% confidence interval is pm 1.96std for a Gaussian, which rounded to 2 stdv.

Reply:
What I mean is that we compute `omega_invsqrt_diag = omega_diag ** (-0.5)` line 274 and line 275 we use it for dividing the quantile.
I don't understand why we don't do a simple multiplication by `omega_diag`.

Reply:
@bthirion

Reply:
The confidence interval is inverse proportional to `omega_diag`, so we need to have one (and only one) inversion. Does that answer your question ?

Reply:
yes

Reply:
@bthirion

Reply:
done